pkg/sentry/config/config.go

```diff
@@ -84,13 +84,21 @@ func FromConfigName(configName string) (conf Config, err error) {
 	if err != nil {
 		err = fmt.Errorf("loading default config. couldn't find config name %q: %w", configName, err)
 		conf = getDefaultConfig()
+		conf.Mode = modes.DaprMode(mode)
 	}
 
 	return conf, err
 }
 
-func IsKubernetesHosted() bool {
-	return os.Getenv(kubernetesServiceHostEnvVar) != ""
+func IsKubernetesHosted(mode modes.DaprMode) bool {
+	switch mode {
+	case modes.KubernetesMode:
+		return true
+	case modes.StandaloneMode:
+		return false
+	default:
+		return os.Getenv(kubernetesServiceHostEnvVar) != ""
```

Comment:
Why do we need this?

Reply:
Added as a safeguard when no mode is specified. In Kubernetes mode, this should never happen since the mode is enforced in opts, but I believe it's not enforced in self-hosted mode. Therefore, I maintained the existing behavior.